qa/tasks/thrashosds.py

```diff
@@ -205,6 +205,7 @@ def task(ctx, config):
         config,
         logger=log.getChild('thrasher')
         )
+    ctx.ceph[cluster].thrashers.append(thrash_proc)
```

Comment:
This class isn't defining the `e` attribute:

````
2019-07-31T12:03:28.450 ERROR:tasks.daemonwatchdog.daemon_watchdog:exception:
Traceback (most recent call last):
  File "/home/teuthworker/src/git.ceph.com_ceph-c_wip-pdonnell-testing-20190730.205527/qa/tasks/daemonwatchdog.py", line 38, in _run
    self.watch()
  File "/home/teuthworker/src/git.ceph.com_ceph-c_wip-pdonnell-testing-20190730.205527/qa/tasks/daemonwatchdog.py", line 109, in watch
    if thrasher.e is not None:
AttributeError: Thrasher instance has no attribute 'e'
````

/ceph/teuthology-archive/pdonnell-2019-07-31_00:40:38-kcephfs-wip-pdonnell-testing-20190730.205527-distro-basic-smithi/4166554/teuthology.log

Suggest that you also create a thrasher super class that allows getting/setting the exception rather than fetching it from this magic and poorly named attribute (my fault!).

Reply:
I thought I fixed this issue. Let me check again.

Reply:
@batrick I see the problem now. I had fixed this only in mon_thrash.py. Need to fix this in other thrashers too. Wondering why this didn't show up in my testing though.

> Suggest that you also create a thrasher super class that allows getting/setting the exception rather than fetching it from this magic and poorly named attribute (my fault!).

I will create a new Base class for this.

Reply:
@batrick Implemented `Thrasher` base class for the thrashers to use. The new Thrasher class should be abstract with an abstract function `_do_thrash`. But that is not within the scope of this PR. So now it sets and gets the exception only.

Reply:
@batrick I have a tracker ticket for it https://tracker.ceph.com/issues/41133.

Reply:
@joscollin I want this to be closed out ASAP. Please just add the `e` attribute so we can get this merged and then we'll fix it later. Please also run this through fs/multimds suites for some small subset (maybe `1/8`) to verify there are no issues.

Reply:
@batrick I have already tested and pushed the code last week. https://github.com/ceph/ceph/pull/28378/commits/21966730dc0c8cb8b0baa19cdc1cf8eb7a3bd5dd.

Reply:
Got it, I'll run it through QA then. Thanks.